### PR TITLE
Chain test

### PR DESCRIPTION
**Pull request by zhukovra**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #105 

Chain test isn't passing tests.

`\Yiisoft\Yii\Web\Middleware\Chain` description is:

> Chain executes middlewares in the order they are added to constructor.

But actually, Chain process only first middleware. Is it bug in Chain class or in the test?

@samdark can you help with clarifying?

### CODE REVIEW

File: tests/Middleware/ChainTest.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+class ChainTest extends TestCase
+{
+    public function testProcess()
+    {
+        $request = $this->createMock(ServerRequestInterface::class);
+        $handler = $this->createMock(RequestHandlerInterface::class);
+
+        $middleware1 = $this->createMock(MiddlewareInterface::class);
+        $middleware1->expects($this->once())->method('process');
```
Comment (zhukovra) [inline tests/Middleware/ChainTest.php]: This executed.

File: tests/Middleware/ChainTest.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+        $middleware2 = $this->createMock(MiddlewareInterface::class);
+        $middleware2->expects($this->once())->method('process');
```
Comment (zhukovra) [inline tests/Middleware/ChainTest.php]: This **not** executed.

**Comment by samdark** [comment]
I guess that is because these are mock middleware. In order for next middleware to be executed, currently executing middlware should delegate handling:

https://github.com/yiisoft/docs/blob/master/guide/en/structure/middleware.md#delegating-handling-to-the-next-middleware

**Comment by zhukovra** [comment]
Thank you for explain. Is test correct now?

**Comment by samdark** [comment]
Merged. Thanks!

**Pull request merged**